.buildkite/scripts/serverless/create_deploy_tag/release_events.ts

```diff
@@ -0,0 +1,312 @@
+      const stateStatus = stateData[stateName];
+      const stateEmoji = {
+        ok: ':white_check_mark:',
+        nok: ':x:',
+        pending: ':hourglass_flowing_sand:',
+        missing: ':white_circle:',
+      }[stateStatus || 'missing'];
+
+      if (stateStatus === 'pending') {
+        return `<div>[${stateEmoji}] ${stateInfo.name}<br />&nbsp; - ${stateInfo.description}</div>`;
+      } else {
+        return `<div>[${stateEmoji}] ${stateInfo.name}</div>`;
+      }
+    });
+
+  const wizardHtml = `<section>
+${wizardHeader}
+${wizardSteps.join('\n')}
+</section>`;
+
+  buildkite.setAnnotation(WIZARD_CTX_DEFAULT, 'info', wizardHtml);
+}
+
+function updateWizardInstruction(targetState: string, stateData: any) {
+  const { instructionStyle, instruction } = states[targetState];
+
+  if (instruction) {
+    buildkite.setAnnotation(
+      WIZARD_CTX_INSTRUCTION,
+      instructionStyle || 'info',
+      `<strong>${instruction}</strong>`
+    );
+  }
+}
+
+async function tryCall(fn: any, ...args: any[]) {
+  if (typeof fn === 'function') {
+    try {
+      const result = await fn(...args);
+      return result !== false;
+    } catch (error) {
+      console.error(error);
+      return false;
+    }
+  } else {
+    return true;
+  }
+}
+
+function sendSlackAnnouncement(selectedCommit: string | null, deployTag: string | null) {
+  const isDryRun = process.env.DRY_RUN?.match('(1|true)');
+  const textBlock = (...str: string[]) => ({ type: 'mrkdwn', text: str.join('\n') });
+  const buildShortname = `kibana-serverless-release #${process.env.BUILDKITE_BUILD_NUMBER}`;
+  sendSlackMessage({
+    blocks: [
+      {
+        type: 'section',
+        text: textBlock(
+          `:ship: Promotion of a new <https://github.com/elastic/kibana/commit/${selectedCommit}|commit> to QA has been initiated!`,
+          `:mag: The details of the candidate selection can be found here: <${process.env.BUILDKITE_BUILD_URL}|${buildShortname}>`,
+          `:test_tube: Once promotion is complete, please begin any required manual testing.`,
+          `*Remember:* Promotion to Staging is currently a manual process and will proceed once the build is signed off in QA.`,
+          `${isDryRun ? '*:white_check_mark:This is a dry run, no action will be taken.*' : ''}`
+        ),
+      },
+      {
+        type: 'section',
+        fields: [
+          textBlock(
+            `*More detail on the candidate selection:*`,
+            `<${process.env.BUILDKITE_BUILD_URL || 'about://blank'}|${buildShortname}>`
+          ),
+          textBlock(`*Initiated by:*`, `${process.env.BUILDKITE_BUILD_CREATOR || 'unknown'}`),
+          textBlock(
+            `*Git tag:*`,
+            `<https://github.com/elastic/kibana/releases/tag/${deployTag}|${deployTag}>`
+          ),
+          textBlock(
+            `*QA Deploy job:*`,
+            `<https://buildkite.com/elastic/kibana-serverless-release/builds?branch=${deployTag}|Link>`
+          ),
```

Comment:
These text blocks don't look great if there's no information below the headline but instead just "Link". They are meant mostly for showing information at a glance (that it can be a link is not the main selling point IMO).

That's why I had the "Useful links" bullet list below in the original message. Then what you set as the headline here could be the linked text. That scales better as it can contain a lot of links easily

Reply:
Unfortunately, we can't ensure that the job exists at the time of messaging, so we can't point to a direct run (which would have a meaningful name) - so it's either this, or I can remove this block. I'll remove it for now. 

In the future, I think we can straight up trigger the release job from this job, instead of setting a tag and waiting for the tag to be parsed/triggered.

Reply:
Removed this